utils/validator_info.go

```diff
@@ -0,0 +1,240 @@
+package utils
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/chain4travel/magellan/cfg"
+	"github.com/chain4travel/magellan/models"
+)
+
+func PeerIndex(peers *models.PeersResponse, nodeID string) int {
+	for idx, peer := range peers.Result.Peers {
+		if peer.NodeID == nodeID {
+			return idx
+		}
+	}
+	return -1
+}
+func GetDate(unixTime string) (string, error) {
+	// Date in Unix Format
+	unixDateInt, err := strconv.ParseInt(unixTime, 10, 64)
+	if err != nil {
+		return "", err
+	}
+	// UnixDate in Time format struct
+	dateFTime := time.Unix(unixDateInt, 0)
+	return strings.Split(dateFTime.String(), " -")[0], nil
+}
+
+func getDuration(startTime string, endTime string) (string, error) {
+	const d = " Days"
+	unixDateInt, err := strconv.ParseInt(startTime, 10, 64)
+	if err != nil {
+		return "", err
+	}
+	start := time.Unix(unixDateInt, 0)
+	unixDateInt, err = strconv.ParseInt(endTime, 10, 64)
+	if err != nil {
+		return "- " + d, err
+	}
+	end := time.Unix(unixDateInt, 0)
+	if err != nil {
+		return "- " + d, err
+	}
+	difference := end.Sub(start)
+	duration := int(difference.Hours() / 24)
+	return strconv.Itoa(duration) + d, nil
+}
+
+func GetValidatorsGeoIPInfo(rpc string, geoIPConfig *cfg.EndpointService) (models.GeoIPValidators, error) {
+	var validatorList []*models.Validator
+	geoValidatorsInfo := &models.GeoIPValidators{
+		Name: "GeoIPInfo",
+	}
+	var errGeoIP error
+	validators, err := GetCurrentValidators(rpc)
+	if err != nil {
+		geoValidatorsInfo.Value = []*models.Validator{}
+		return *geoValidatorsInfo, err
+	}
+	peers, _ := GetPeers(rpc)
+	if !reflect.DeepEqual(validators, models.ValidatorsResponse{}) && !reflect.DeepEqual(peers, models.PeersResponse{}) &&
```

Comment:
Suggestion:
Line 57: var validatorList []*models.Validator{}
then remove the complete huge (and expensive) if statement (beside the fact that the latter 2 conditions are not really required)

Side note: In real life (multiple nodes) you'll run into the issue that you get a validator but incomplete peers list.
Every node stores only a bunch of peers, but in general not all (only neighbors)